lib/services/elasticsearch.js

```diff
@@ -972,7 +945,7 @@ class ElasticSearch extends Service {
     // fix #1131: we should ignore the "collection" argument even if
     // one is provided, as "listing the provided collection name"
     // makes no sense
-    delete esRequest.type;
+    delete collection;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this "delete" operator or pass an object property to it. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS3001)